appendix/single-sign-on.rst

```diff
@@ -0,0 +1,521 @@
+
+:the secret key:                in parentheses at the end (**0x5ee827...**)
+:the secret key version number: preceded by ``vno`` (**3**)
+
+Step 2: Remove NGINX, Set up Apache + Kerberos
+----------------------------------------------
+
+Next, the Zammad host must be configured to support Kerberos
+(and to accept auth credentials provided by the Active Directory server).
+
+In most cases, you would have to recompile NGINX from source
+with an extra module to enable Kerberos support.
+To get around this, we will use Apache,
+which offers Kerberos support through a plug-in module instead.
+
+.. note:: All commands in this section must be run as root (or with ``sudo``).
+
+2a. Turn off NGINX
+^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   $ systemctl stop nginx     # turn off nginx
+   $ systemctl disable nginx  # keep it off after reboot
+
+.. warning:: This will take your Zammad instance **offline**
+   until Apache is fully configured and running.
+
+   If you wish to minimize downtime, you can save this step for last;
+   just bear in mind that Apache will not start
+   if the port it wants to listen on is being used by NGINX.
+
+   If for any reason you can’t complete this tutorial,
+   simply turn off Apache and restore NGINX:
+
+   .. code-block:: sh
+
+      $ systemctl stop apache2
+      $ systemctl disable apache2
+      $ systemctl enable nginx
+      $ systemctl start nginx
+
+2b. Install Apache
+^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   # Ubuntu & Debian
+   $ apt update
+   $ apt install apache2 krb5-user libapache2-mod-auth-kerb
+
+   # CentOS
+   $ yum install httpd krb5-workstation mod_auth_kerb
+
+   # openSUSE
+   $ zypper ref
+   $ zypper install apache2 krb5-client apache2-mod_auth_kerb
+
+2c. Enable Apache modules
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   # Ubuntu, Debian, & openSUSE
+   $ a2enmod auth_kerb headers rewrite proxy proxy_html proxy_http proxy_wstunnel
+
+On systems without ``a2enmod`` (*e.g.,* CentOS),
+add/uncomment the appropriate ``LoadModule`` statements
+in your Apache config:
+
+.. code-block::
+
+   # /etc/httpd/conf/httpd.conf
+
+   LoadModule auth_kerb_module /usr/lib/apache2/modules/mod_auth_kerb.so
+   LoadModule headers_module modules/mod_headers.so
+   LoadModule rewrite_module modules/mod_rewrite.so
+   LoadModule proxy_module modules/mod_proxy.so
+   LoadModule proxy_html_module modules/mod_proxy_html.so
+   LoadModule proxy_http_module modules/mod_proxy_http.so
+   LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so
+
+2d. Configure Kerberos
+^^^^^^^^^^^^^^^^^^^^^^
+
+Kerberos realm configuration is how you tell the Zammad server
+about the *domain controller* (Active Directory server).
+
+.. note:: Replace the following placeholders in the sample config below:
+
+   :``<domain>``:                   Windows domain
+   :``<domain-controller>``:        Domain controller IP/FQDN(s)
+   :``<master-domain-controller>``: Master domain controller IP/FQDN
```

Comment:
I don't know anything about Windows system administration, so I was really guessing at what these placeholders were supposed to represent.

Did I get it right?

Reply:
Quite on point!
The realm configuration is about how your server is to reach the responsible domain controllers. I think your leading in sentence fits with that.

One thing that concerns me is actually the `admin_server` and `kdc` part.
Originally the file said this:
```
                      # you can define more than one kdc (each on it's own line)	
                      # this allows you to provide secondaries if needed	
                      kdc = {IP / FQDN of domain controller}	
                      # admin_server can be the same as kdc if it's not read only	
                      admin_server = {IP / FQDN of master domain controller}
```

Which actually is a hint on how to configure the file. We may thus want to add it above or below the sample.
Per `kdc` line you may only provide one domain controller. In terms of KDC it doesn't matter if the domain controller you define here is read only (this part is not relevant in my opinion)
The `admin_server` *can* be the same as one of the above mentioned kdcs. This may be true if you e.g. only use one domain controller. Not sure if that helps.

Reply:
* “Per `kdc` line you may only provide one domain controller. In terms of KDC it doesn't matter if the domain controller you define here is read only (this part is not relevant in my opinion)”

  Tried to cover that in [this line](https://github.com/zammad/zammad-documentation/pull/147/files#diff-27d6b2138d1c132cf617d7a5301c17a0R282), but happy to rephrase it as:

  ```
  # multiple KDCs ok (one `kdc = ...` statement per line)
  ```

* The `admin_server` can be the same as one of the above mentioned kdcs. This may be true if you e.g. only use one domain controller.

  This is covered in [these lines](https://github.com/zammad/zammad-documentation/pull/147/files#diff-27d6b2138d1c132cf617d7a5301c17a0R262-R263), which appear as part of the definition list item:

  ![Screenshot_20201002_102807](https://user-images.githubusercontent.com/12194123/94882267-0f1d8680-049a-11eb-8838-4e497dd3ba79.png)

Thoughts?

Reply:
Okay, let's try it that way then and see what happens.